game/core/views/animated_sprite.py

```diff
@@ -0,0 +1,51 @@
+"""Class for animated sprite."""
+
+import arcade
+from beartype import beartype
+
+
+class AnimatedSprite(arcade.Sprite):
+    @beartype
+    def __init__(
+        self, game_clock, name, center_x, center_y, paired_sprite=None, scale=1
+    ) -> None:
+        super().__init__()
+
+        # Configure when to update texture
+        self.game_clock = game_clock
+        self.update_time = 0.2
+        self.cur_texture_index = 1
+        self.increase_index = True
+        self.time_to_update_texture = self.game_clock.get_time_in_future(
+            self.update_time
+        )
+
+        # Configure sprite
+        self.paired_sprite = paired_sprite
+        self.name = name
+        self.center_x = center_x
+        self.center_y = center_y
+        self.scale = scale
+        self.texture = arcade.load_texture(self.get_filename())
+
+    @beartype
+    def get_filename(self) -> str:
+        return f":misc:{self.name}-{self.cur_texture_index}.png"
+
+    @beartype
+    def on_update(self, delta_time) -> None:
+        if self.paired_sprite and "removed" in self.paired_sprite.properties:
+            self.remove_from_sprite_lists()
+        if self.game_clock.current_time > self.time_to_update_texture:
+            if self.cur_texture_index == 3:
+                self.increase_index = False
```

Comment:
With the proposed file restructure, you could get the number of indices based on a glob search of the `sparkle` directory, but it could just be a constant argument

Then, you could simplify this code block with with something like `self.index_counter = itertools.cycle(range(1, steps + 1))` and called with `self.cur_texture_index = next(self.index_counter)`

Reply:
We can definitely refactor this once we have animated assets that require different number of frames. 
3 frames is a sweet spot because it has decent animation results while not requiring too much effort/time in creating the frames.

Reply:
I made the changes to the naming and the nested structure. Going to merge this since you've approved it!